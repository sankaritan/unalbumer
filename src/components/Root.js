import React, { Component } from "react";
import { connect } from "react-redux";
import { GoogleLogin } from "react-google-login";
import { loginUserAction } from "../actions";
import PhotosWrapper from "./PhotosWrapper";
import PageHeader from './PageHeader';
import PropTypes from "prop-types";

class Root extends Component {
  loginSuccess = (response) => {
    const username = response.profileObj.name;
    const token = response.accessToken;
    this.props.dispatch(loginUserAction(username, token));
  };

  loginFailure = () => {
    // TODO handle Google login failure
  };

  render() {
    return (
      <div>
          <PageHeader/>
        {this.props.loggedIn ? (
          <div>
            <p>Logged in as {this.props.username}!</p>
            <PhotosWrapper />
          </div>
        ) : (
          <div>
            <GoogleLogin
              clientId="493662417086-dsm3ap89a5uvlv28sn9f7qu19fi7k26p.apps.googleusercontent.com"
              scope="https://www.googleapis.com/auth/photoslibrary"
              onSuccess={this.loginSuccess}
              onFailure={this.loginFailure}
              className={ 'button' }
            />
            <div id="google-login" />
          </div>
        )}
      </div>
    );
  }
}

Root.propTypes = {
  dispatch: PropTypes.func,
  username: PropTypes.string,
  loggedIn: PropTypes.bool
};

const mapStateToProps = (state) => ({
  username: state.user.username,
  loggedIn: state.user.loggedIn
});

export default connect(mapStateToProps)(Root);
