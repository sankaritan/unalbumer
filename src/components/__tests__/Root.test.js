import React from "react";
import { Root } from "../Root";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

describe("Root component", () => {
  const testOptions = [
    {
      dispatch: jest.fn(),
      username: "jardajagr",
      loggedIn: true
    },
    {
      dispatch: jest.fn(),
      username: null,
      loggedIn: false
    }
  ];

  testOptions.forEach((props) => {
    it("Root renders correctly", () => {
      const component = shallow(<Root {...props} />).debug();
      expect(component).toMatchSnapshot();
    });
  });
});
