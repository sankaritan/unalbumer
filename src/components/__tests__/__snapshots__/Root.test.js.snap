// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component Root renders correctly 1`] = `
"<div>
  <Header />
  <div>
    <p>
      Logged in as 
      jardajagr
      !
    </p>
    <Connect(Photos) />
  </div>
</div>"
`;

exports[`Root component Root renders correctly 2`] = `
"<div>
  <Header />
  <div>
    <div>
      <t clientId=\\"493662417086-dsm3ap89a5uvlv28sn9f7qu19fi7k26p.apps.googleusercontent.com\\" scope=\\"https://www.googleapis.com/auth/photoslibrary\\" onSuccess={[Function]} onFailure={[Function]} className=\\"button\\" type=\\"button\\" tag=\\"button\\" buttonText=\\"Login with Google\\" accessType=\\"online\\" prompt=\\"\\" cookiePolicy=\\"single_host_origin\\" fetchBasicProfile={true} isSignedIn={false} uxMode=\\"popup\\" disabledStyle={{...}} onRequest={[Function: onRequest]} jsSrc=\\"https://apis.google.com/js/client:platform.js\\" />
      <div id=\\"google-login\\" />
    </div>
  </div>
</div>"
`;
